### PR TITLE
Add the IEdmOperationReturnType interface and enable annotation on return type of operation

### PR DESCRIPTION
**Pull request by xuzhg**
### Issues

*This pull request fixes issue #52.*

### Description

* Introduce the `IEdmOperationReturnType` interface
* Modify the `IEdmOperation` to use `IEdmOperationReturnType` interface.
* Enable in-line and out of line annotation on return type.

### Checklist (Uncheck if it is not completed)

- [x] *Test cases added*
- [x] *Build and test with one-click build and test script passed*

### Additional work necessary

*If documentation update is needed, please add "Docs Needed" label to the issue and provide details about the required document change in the issue.*

### CODE REVIEW

File: src/Microsoft.OData.Edm/Csdl/Parsing/Ast/CsdlOperationReturnType.cs [hunk 1]
```diff
@@ -4,26 +4,31 @@
...
-            this.returnType = returnType;
+            this.ReturnType = returnType;
+        }
+
+        public CsdlTypeReference ReturnType { get; }
+
+        public override void AddAnnotation(CsdlDirectValueAnnotation annotation)
+        {
+            base.AddAnnotation(annotation);
+            ReturnType.AddAnnotation(annotation);
```
Comment (mikepizzo) [inline src/Microsoft.OData.Edm/Csdl/Parsing/Ast/CsdlOperationReturnType.cs]: Is this right? Should we be annotating both the OperationReturnType and the CsdlTypeReference?  I would have expected that we only annotate the CsdlOperationReturnType...

File: src/Microsoft.OData.Edm/Csdl/Parsing/Ast/CsdlOperationReturnType.cs [hunk 2]
```diff
@@ -4,26 +4,31 @@
...
         }
 
-        public CsdlTypeReference ReturnType
+        public override void AddAnnotation(CsdlAnnotation annotation)
         {
-            get { return this.returnType; }
+            base.AddAnnotation(annotation);
+            ReturnType.AddAnnotation(annotation);
```
Comment (mikepizzo) [inline src/Microsoft.OData.Edm/Csdl/Parsing/Ast/CsdlOperationReturnType.cs]: As above; should we be annotating *both* the CsdlTypeReference and the CsdlOperationReturnType?

File: src/Microsoft.OData.Edm/Csdl/Parsing/Ast/CsdlOperationReturnType.cs (hunk 1 above)
Comment (xuzhg) [reply src/Microsoft.OData.Edm/Csdl/Parsing/Ast/CsdlOperationReturnType.cs]: I have to do like this, and I double check that's ok. 
The reason is that:

1. Only the "CsdlTypeReference" will be used to constructor the "CsdlSemanticsOperation....ReturnType"'.

2. CsdlOperationReturnType is only a middle wrapper class, it's not used in later to constructor the "CsdlSemanticsOperation....ReturnType"'.

File: src/Microsoft.OData.Edm/Csdl/Parsing/Ast/CsdlOperationReturnType.cs (hunk 2 above)
Comment (xuzhg) [reply src/Microsoft.OData.Edm/Csdl/Parsing/Ast/CsdlOperationReturnType.cs]: Basically, we should annotate "CsdlOperationReturnType" only, however, we use the "CsdlTypeReference returnType" to constructor the other ""CsdlSemanticsOperation....ReturnType"'. 

So, I have to do like this. Of course, we can use "base.AddAnnotation(annotation), but for safety, it 's better to keep the base call.

**Comment by xuzhg** [comment]
Close this because the #1414 enables this functionality. Thanks.

**Pull request closed without merging**